### PR TITLE
Add support for Breadcrumbs and Custom Code Language to Shnip

### PR DESCRIPTION
**Pull request by acekyd**
## What type of PR is this? (check all applicable)

- [x] ♻️ Refactor
- [x] ✨ New Feature

## Description

This PR adds support for Breadcrumbs through `nestedSnippets` and also allows specifying custom code language to use for imported snippets e.g rendering bash under the JS tab. 

In use here #1232 

## Related Tickets & Documents

Resolves #

## Added code snippets?
- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 readme
- [ ] 📜 contributing.md
- [ ] 📓 general documentation
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: site/src/components/language/Shnip.jsx [hunk 1]
```diff
@@ -48,7 +48,7 @@ const Shnip = ({ snippets, inlineSnippets }) => {
                 {!nestedSnippets &&
                   <CodeSnippet
                     snippet={snippetContent}
-                    language={codeLanguage.toLowerCase() || language.toLowerCase()}
```
Comment (blackgirlbytes) [inline site/src/components/language/Shnip.jsx]: wait why remove the toLowerCase part? I think I added that, but I don't remember why I did 😅

**Comment by blackgirlbytes** [comment]
okay, awesome! Glad you added this so that the language switcher in the other PR will actually work!

**Review by blackgirlbytes — APPROVED** [review]

File: site/src/components/language/Shnip.jsx (hunk 1 above)
Comment (acekyd) [reply site/src/components/language/Shnip.jsx]: Oh I added it here and for some reason it was breaking the build. The one you added is still below and works with the build.

**Review by angiejones — CHANGES_REQUESTED** [review]
- Missing the dropdown on the pages
- When switching one tab to Kotlin, the other does not switch

https://deploy-preview-1233--tbd-website-developer.netlify.app/docs/tbdex/pfi/required-sdks?lang=Kotlin

image

File: site/src/components/language/Shnip.jsx (hunk 1 above)
Comment (blackgirlbytes) [reply site/src/components/language/Shnip.jsx]: oh okay!

**Comment by blackgirlbytes** [comment]
> * Missing the dropdown on the pages
> * When switching one tab to Kotlin, the other does not switch
> 
> https://deploy-preview-1233--tbd-website-developer.netlify.app/docs/tbdex/pfi/required-sdks?lang=Kotlin
> 
> image

Oh I was thinking if he merged this, and then that would enable him to pull in these changes to his other PR that adds the  global language switching tab

**Comment by acekyd** [comment]
Yes @blackgirlbytes. I wanted to keep this PR clean and focus on just the breadcrumbs update. And then #1232 would make the changes once this is merge. But I'll just combine everything here to make it easier since they are connected. @angiejones

**Review by chris-tbd — APPROVED** [review]

**Review by dayhaysoos — APPROVED** [review]

**Review by angiejones — APPROVED** [review]

**Comment by angiejones** [comment]
Ace thank you so much for cleaning this up! much appreciated 🙏🏾

**Comment by angiejones** [comment]
> Yes @blackgirlbytes. I wanted to keep this PR clean and focus on just the breadcrumbs update. And then #1232 would make the changes once this is merge. But I'll just combine everything here to make it easier since they are connected. @angiejones

Oh I'm sorry y'all. I didn't realize there were 2 PRs

**Comment by acekyd** [comment]
Haha no worries, Angie! <3

**Pull request merged**